components/Dialogs/src/private/MDCAlertActionManager.m

```diff
@@ -0,0 +1,95 @@
+// Copyright 2018-present the Material Components for iOS authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import "MDCAlertActionManager.h"
+
+@interface MDCAlertActionManager ()
+@property(nonatomic, nonnull, strong) NSMapTable *actionButtons;
+@end
+
+@implementation MDCAlertActionManager {
+  NSMutableArray<MDCAlertAction *> *_actions;
+}
+
+@dynamic sortedButtons;
+
+- (instancetype)init {
+  self = [super init];
+  if (self) {
+    _actions = [[NSMutableArray alloc] init];
+    _actionButtons = [NSMapTable mapTableWithKeyOptions:NSMapTableWeakMemory
```

Comment:
Do we need the weak keys for our action items? If not, I recommend replacing with NSMutableDictionary.

Reply:
We don't necessarily need weak keys, however NSDictionary copies the keys (the actions in this case) which makes it harder to later access an action with a subscript. 
Alternatively, I could store the action objects in an NSDictionary as an NSValue. 
Not sure which solution is better. Any thoughts?

Reply:
Ah - you may want to specify NSPointerFunctionsOpaquePersonality as well then to indicate that you're not implementing -hash and -isEqual to result in two copies of the same data equating to one another.

Reply:
Speaking offline, we looked at this and felt it may be simpler to just have one array with an NSObject that stores an action and button property, something like so:

```objc
interface ActionUserInfo: NSObject {
  initWithAction:
  property(readonly) action;
  property(readwrite nullable) button;
}
```

Reply:
Followup discussion: https://github.com/material-components/material-components-ios/issues/5447